### PR TITLE
feat(ui5-side-navigation): add navigation groups

### PR DESCRIPTION
**Pull request by TeodorTaushanov**

### CODE REVIEW

**Review by s-todorova — CHANGES_REQUESTED** [review]
When `ui5-side-navigation-group` or `ui5-side-navigation-item` with subitem children has the `disabled` property set, it shouldn't be able to be pressed or selected. Currently, disabled group or parent items are not focusable/in the tab chain, but they can still be pressed and trigger their expand/collapse. Instead, such items/groups should behave as in ui5 - making all their child items disabled as well and should not be interactive (see "Restricted" group in [this sample](Restricted)).

File: packages/playground/_stories/fiori/SideNavigation/SideNavigationGroup/SideNavigationGroup.stories.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	component: "SideNavigationGroup",
+	argTypes,
+} as Meta<SideNavigationGroup>;
+
+const Template: UI5StoryArgs<SideNavigationGroup, StoryArgsSlots> = (args) => {
+	return html`
+<ui5-side-navigation>
+	<ui5-side-navigation-group
+		text="${ifDefined(args.text)}"
+		?expanded="${ifDefined(args.expanded)}"
```
Comment (s-todorova) [inline packages/playground/_stories/fiori/SideNavigation/SideNavigationGroup/SideNavigationGroup.stories.ts:22]: `?disabled="${ifDefined(args.disabled)}"` and below in `Basic.args` missing, control in storybook for this property is not working,
same for SideNavigationItem.stories.ts and SideNavigationSubItem.stories.ts

File: packages/playground/_stories/fiori/SideNavigation/SideNavigationGroup/SideNavigationGroup.stories.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		${unsafeHTML(args.default)}
+	</ui5-side-navigation-group>
+</ui5-side-navigation>`;
+};
+
+export const Basic = Template.bind({});
+Basic.tags = ["_hidden_"];
+Basic.args = {
+	text: "Group",
+	expanded: true,
```
Comment (s-todorova) [inline packages/playground/_stories/fiori/SideNavigation/SideNavigationGroup/SideNavigationGroup.stories.ts:33]: add `disabled=false,`

**Review by s-todorova — CHANGES_REQUESTED** [review]

**Review by GerganaKremenska — CHANGES_REQUESTED** [review]
1. When collapsed is false, SideNavigation conatiner padding: 1.5rem 0.75rem 0.75rem 0.75rem;

2. Should  space on group, expand it and collapse it?

**Comment by TeodorTaushanov** [comment]
> 1. When collapsed is false, SideNavigation conatiner padding: 1.5rem 0.75rem 0.75rem 0.75rem;
> 2. Should  space on group, expand it and collapse it?


1. From the spec 
-  Desktop Specification (Compact) - padding: 1.5rem 0.75rem 0.75rem 0.75rem; 
-  Tablet Specification (Cozy) - padding: 0.5rem;

2. Collapse/Expand via Space and Enter key is removed.

File: packages/fiori/src/SideNavigationSubItem.hbs [hunk 3]
```diff
@@ -0,0 +1,57 @@
+{{#unless sideNavCollapsed}}
+	<li id="{{_id}}" class="ui5-sn-list-li" role="none">
+		{{#if _href}}
+			<a class="ui5-sn-item ui5-sn-item-level2 {{_classes}}"
+			   role="treeitem"
+			   data-sap-focus-ref
```
[the remark this answers is not in the document]
Comment (TeodorTaushanov) [reply packages/fiori/src/SideNavigationSubItem.hbs:6]: Added aria-description= "Group Header"
"aria-level" is not needed as JAWS and NVDA calculate the levels automatically.

**Comment by TeodorTaushanov** [comment]
> When `ui5-side-navigation-group` or `ui5-side-navigation-item` with subitem children has the `disabled` property set, it shouldn't be able to be pressed or selected. Currently, disabled group or parent items are not focusable/in the tab chain, but they can still be pressed and trigger their expand/collapse. Instead, such items/groups should behave as in ui5 - making all their child items disabled as well and should not be interactive (see "Restricted" group in [this sample](Restricted)).

This should be discussed with the interaction designers.

File: packages/fiori/src/SideNavigationGroup.ts [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+// Styles
+import SideNavigationItemCss from "./generated/themes/SideNavigationItem.css.js";
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * Represents a group of navigation actions.
```
Comment (nnaydenow) [inline packages/fiori/src/SideNavigationGroup.ts:31]: switch to markdown format for JSDOC and merge main

Comment (TeodorTaushanov) [reply packages/fiori/src/SideNavigationGroup.ts:31]: Done.

**Comment by TeodorTaushanov** [comment]
> > When `ui5-side-navigation-group` or `ui5-side-navigation-item` with subitem children has the `disabled` property set, it shouldn't be able to be pressed or selected. Currently, disabled group or parent items are not focusable/in the tab chain, but they can still be pressed and trigger their expand/collapse. Instead, such items/groups should behave as in ui5 - making all their child items disabled as well and should not be interactive (see "Restricted" group in [this sample](Restricted)).
> 
> This should be discussed with the interaction designers.

The "disabled behavior" will be fixed with another PR.

File: packages/fiori/src/SideNavigationGroup.ts [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+// Styles
+import SideNavigationItemCss from "./generated/themes/SideNavigationItem.css.js";
+
+/**
+ * @class
+ *
+ * ### Overview
+ *
+ * Represents a group of navigation actions.
+ * The `ui5-side-navigation-group` can to be used inside a `ui5-side-navigation` only.
```
Comment (BorisDafov) [inline packages/fiori/src/SideNavigationGroup.ts:32]: Remove or if it's important to remain:
The `ui5-side-navigation-group` can only be used inside a `ui5-side-navigation`.

File: packages/fiori/src/SideNavigationGroup.ts [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+// Styles
+import SideNavigationItemCss from "./generated/themes/SideNavigationItem.css.js";
+
+/**
+ * @class
+ *
+ * ### Overview
+ *
+ * Represents a group of navigation actions.
```
Comment (BorisDafov) [inline packages/fiori/src/SideNavigationGroup.ts:31]: I suggest:
Represents a group of navigation actions within `ui5-side-navigation`.

File: packages/fiori/src/SideNavigationSelectableItemBase.ts [hunk 7]
```diff
@@ -0,0 +1,173 @@
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import event from "@ui5/webcomponents-base/dist/decorators/event.js";
+import { isSpace, isEnter } from "@ui5/webcomponents-base/dist/Keys.js";
+import SideNavigationItemBase from "./SideNavigationItemBase.js";
+
+/**
+ * Fired when the component is activated either with a
+ * click/tap or by using the Enter or Space key.
```
Comment (BorisDafov) [inline packages/fiori/src/SideNavigationSelectableItemBase.ts:9]: click/tap or by using the [Enter] or [Space] keys.

File: packages/fiori/src/SideNavigationSubItem.ts [hunk 8]
```diff
@@ -1,28 +1,40 @@
...
+
+// Styles
+import SideNavigationItemCss from "./generated/themes/SideNavigationItem.css.js";
 
 /**
  * @class
  *
  * ### Overview
- *
+ * Represents a single navigation action.
```
Comment (BorisDafov) [inline packages/fiori/src/SideNavigationSubItem.ts:15]: I suggest:
Represents a single navigation action within `ui5-side-navigation-item`.

File: packages/fiori/test/pages/SideNavigationWithGroups.html [hunk 9]
```diff
@@ -0,0 +1,120 @@
+<!DOCTYPE html>
+<html>
+<head>
+	<meta charset="utf-8">
+	<title>Side Navigation With Groups</title>
```
Comment (BorisDafov) [inline packages/fiori/test/pages/SideNavigationWithGroups.html:5]: <title>Side Navigation with Groups</title>

**Review by BorisDafov — CHANGES_REQUESTED** [review]

File: packages/fiori/test/pages/SideNavigationWithGroups.html (hunk 9 above)
Comment (dimovpetar) [reply packages/fiori/test/pages/SideNavigationWithGroups.html:5]: Done

File: packages/fiori/src/SideNavigationSubItem.ts (hunk 8 above)
Comment (dimovpetar) [reply packages/fiori/src/SideNavigationSubItem.ts:15]: Done

File: packages/fiori/src/SideNavigationSelectableItemBase.ts (hunk 7 above)
Comment (dimovpetar) [reply packages/fiori/src/SideNavigationSelectableItemBase.ts:9]: Done

File: packages/fiori/src/SideNavigationGroup.ts (hunk 6 above)
Comment (dimovpetar) [reply packages/fiori/src/SideNavigationGroup.ts:31]: Done

File: packages/fiori/src/SideNavigationGroup.ts (hunk 5 above)
Comment (dimovpetar) [reply packages/fiori/src/SideNavigationGroup.ts:32]: Did the latter

**Comment by BorisDafov** [comment]
I approve the change from UA perspective....

**Review by GerganaKremenska — APPROVED** [review]

**Pull request merged**